wfe/wfe_test.go

```diff
@@ -1592,3 +1592,46 @@ func TestVerifyPOSTInvalidJWK(t *testing.T) {
 	test.AssertEquals(t, probs.MalformedProblem, prob.Type)
 	test.AssertEquals(t, http.StatusBadRequest, prob.HTTPStatus)
 }
+
+func TestHeaderBoulderRequestId(t *testing.T) {
+	wfe, _ := setupWFE(t)
+	mux, err := wfe.Handler()
+	test.AssertNotError(t, err, "Problem setting up HTTP handlers")
+	responseWriter := httptest.NewRecorder()
+
+	mux.ServeHTTP(responseWriter, &http.Request{
+		Method: "GET",
+		URL:    mustParseURL(directoryPath),
+	})
+
+	test.AssertNotEquals(t, responseWriter.Header().Get("Boulder-Request-ID"), "")
```

Comment:
It would be nice to see a test for the non-empty `Boulder-Request-ID` path as well. It seems like the primary challenge there is that the ID is randomly generated using `core.NewToken` and can't be predicted by the test unless we mock it or use a seeded RNG.

@jsha @benileo Do you think it's worth the trouble of figuring out how to mock out a predictable `core.NewToken` for testing or should we accept as-is with the empty value test?